db/model/PostLink.ts

```diff
@@ -0,0 +1,47 @@
+import { Entity, PrimaryGeneratedColumn, Column, BaseEntity } from "typeorm"
+import { formatUrls } from "../../site/formatting.js"
+import { Url } from "@ourworldindata/utils"
+import { getLinkType, getUrlTarget } from "@ourworldindata/components"
+
+@Entity("posts_links")
+export class PostLink extends BaseEntity {
+    @PrimaryGeneratedColumn() id!: number
+    // TODO: posts is not a TypeORM but a Knex class so we can't use a TypeORM relationship here yet
+
+    @Column({ type: "int", nullable: false }) sourceId!: number
+
+    @Column() linkType!: "gdoc" | "url" | "grapher" | "explorer"
+    @Column() target!: string
+    @Column() queryString!: string
+    @Column() hash!: string
+    @Column() componentType!: string
+    @Column() text!: string
+
+    static createFromUrl({
+        url,
+        sourceId,
+        text = "",
+        componentType = "",
+    }: {
+        url: string
+        sourceId: number
+        text?: string
+        componentType?: string
```

Comment:
Since `text` and `componentType` are both `NOT NULL`, is it valid to have them as optional here?
Will TypeORM be clever about it and turn them into an empty string if missing, or will it fail?

Reply:
Ah I see! I hate how with object destructuring in function parameters, you have to do the destructuring and TS typing separately...